docs/en/get_started/quick_start.md

```diff
@@ -0,0 +1,745 @@
+
+## Creating objects using `rdl!`
+
+`rdl` is the abbreviation of Ribir Declarative Language, and the purpose of the `rdl!` macro is to help you create objects in a declarative way.
+
+> Notice
+>
+> `rdl!` does not care about types, it only does processing at the syntax level, so it is not only widgets that can use it.
+
+### Declarative creation of objects
+
+Although `rdl!` supports any Rust expression, but what we mean by declarative creation of objects, specifically refers to the way of creating objects through structure literals.
+
+When your expression is a structure literal, `rdl!` will create an object through the `Declare` trait, which requires that the type of the object you create must inherit or implement the `Declare` trait.
+
+
+```rust
+use ribir::prelude::*;
+
+#[derive(Declare)]
+pub struct Counter {
+  #[declare(default = 1usize)]
+  count: usize,
+}
+// `rdl!` only allow to be used in a context with `ctx!(): &BuildCtx` accessible.
+// So we use a function with `ctx!()` parameter to provide this context.
+fn use_rdl(ctx!(): &BuildCtx) {
+  let _ = rdl!{ Counter { } };
+}
+```
+
+In the above example, `Counter` inherits `Declare` and marks the default value of `count` as `1`. So in `rdl!`, you don't need to assign a value to `count`, `rdl!` will assign it a default value of `1` when creating it. `Declare` has some other features, which we will not expand here.
+
+## Composing widgets
+
+You already know how to create a widget, and now we will compose a simple counter application by nesting widgets in another widget.
+
+You can nest additional `rdl!` instances as children within the widget declared by the structure literal. Please note that child widgets must always be declared after the parent widget's properties. This is a formatting requirement of `rdl!`.
+
+
+```rust
+use ribir::prelude::*;
+
+fn main() {
+  let counter = fn_widget! { 
+    rdl!{ 
+      Row {
+        rdl!{ FilledButton {
+          rdl! { Label::new("Increment") }
+        }}
+        rdl!{ H1 { text: "0" } }
+      }
+    }
+  };
+
+  App::run(counter);
+}
+```
+
+In the above example, we created a `Row` with two child nodes, `FilledButton` and `H1`. These three widgets are already defined in the `ribir_widgets` library.
+
+`rdl!` also allows you to declare children for widgets that have already been created:
+
+```rust
+use ribir::prelude::*;
+
+fn main() {
+  let counter = fn_widget! {
+    let row = rdl!{ Row { align_items: Align::Center } };
+
+    rdl!{ 
+      $row {
+        rdl!{ FilledButton {
+          rdl! { Label::new("Increment") }
+        }}
+        rdl!{ Text { text: "0" } }
+      }
+    }
+  };
+
+  App::run(counter);
+}
+```
+
+Do you notice the `$row { $row { ... } }`? It is the same as the structure literal syntax, but with `$` in front of it, it means that it is a variable rather than a type, so it will not create a new widget, but directly use this variable to compose with the child.
```

Comment:
Do you notice the `$row { $row { ... } }` => Do you notice the `rdl! { $row { ... } }`